gtsfm/utils/tracks.py

```diff
@@ -0,0 +1,72 @@
+"""Utilities for 2D and 3D tracks.
+
+Authors: Ayush Baid
+"""
+from typing import Dict, List
+
+from gtsam import Cal3Bundler, SfmTrack
+
+from gtsfm.common.sfm_track import SfmMeasurement, SfmTrack2d
+from gtsfm.data_association.point3d_initializer import (
+    Point3dInitializer,
+    TriangulationParam,
+    TriangulationExitCode,
+)
+
+
+def classify_tracks2d_with_gt_cameras(
+    tracks: List[SfmTrack2d], cameras_gt: List[Cal3Bundler], reproj_error_thresh_px: float = 3
+) -> List[TriangulationExitCode]:
+    """Classifies the 2D tracks w.r.t ground truth cameras by performing triangulation and collecting exit codes.
+
+    Args:
+        tracks: list of 2d tracks.
+        cameras_gt: cameras with GT params.
+        reproj_error_thresh_px (optional): Reprojection error threshold (in pixels) for a track to be considered an
+                                           all-inlier one. Defaults to 3.
+
+    Returns:
+        The triangulation exit code for each input track.
+    """
+    # do a simple triangulation with the GT cameras
+    cameras_dict: Dict[int, Cal3Bundler] = {i: cam for i, cam in enumerate(cameras_gt)}
+    point3d_initializer = Point3dInitializer(
+        track_camera_dict=cameras_dict, mode=TriangulationParam.NO_RANSAC, reproj_error_thresh=reproj_error_thresh_px
+    )
+
+    exit_codes: List[TriangulationExitCode] = []
+    for track in tracks:
+        _, _, triangulation_exit_code = point3d_initializer.triangulate(track_2d=track)
+        exit_codes.append(triangulation_exit_code)
+
+    return exit_codes
+
+
+def classify_tracks3d_with_gt_cameras(
+    tracks: List[SfmTrack], cameras_gt: List[Cal3Bundler], reproj_error_thresh_px: float = 3
+) -> List[TriangulationExitCode]:
+    """Classifies the 3D tracks w.r.t ground truth cameras by performing triangulation and collecting exit codes.
+
+    Args:
+        tracks: list of 3d tracks.
+        cameras_gt: cameras with GT params.
+        reproj_error_thresh_px (optional): Reprojection error threshold (in pixels) for a track to be considered an
+                                           all-inlier one. Defaults to 3.
+
+    Returns:
+        The triangulation exit code for each input track.
```

Comment:
is this list the same length as `tracks`? should we call them both J or K or M or N in length?